### ISSUE TITLE
Commenting on big diffs

### ISSUE THREAD

**Issue by dmuzalevskiy** [ReviewNB/support#13]
Getting an error message while commenting on big diffs:

"There was an error in posting your comments. One of your PR file has unusually large diff. Currently Github does not accept review comments on very large diffs. Please reduce diff or remove the comments on that particular file in the PR". 

Can this be solved sometime soon? 

Thanks.

**Comment by amit1rrr** [comment]
@dmuzalevskiy Sorry for the inconvenience. This is a current limitation/bug of GitHub API where it does not accept Review Comment when diff for the file is over 1 mb (limit identified by [dshean](https://github.com/dshean) with some experimentation). 

I opened an [issue](https://github.community/t5/GitHub-API-Development-and/quot-Position-is-invalid-quot-error-in-posting-PR-review-via-v3/m-p/18932#M881) on GitHub community and also talking to their support via email. You can see some replies by GitHub towards the end of that thread ([this](https://github.community/t5/GitHub-API-Development-and/quot-Position-is-invalid-quot-error-in-posting-PR-review-via-v3/m-p/19220/highlight/true#M938) and [this](https://github.community/t5/GitHub-API-Development-and/quot-Position-is-invalid-quot-error-in-posting-PR-review-via-v3/m-p/19644/highlight/true#M987)).

To summarize, GitHub acknowledged it as a bug/limitation on their API but haven't provided any timeline to fix this. I will post further updates here. Workaround is to store comments in ReviewNB itself but I don't think it's a good idea from security/privacy standpoint.

**Comment by amit1rrr** [comment]
I built a workaround that lets user posts comments on any size diffs. [More information here](https://github.com/ReviewNB/support/issues/29) but the gist is that we post review comments as overall PR comments along with context and other metadata. 

I'm particularly proud of this solution since it's been a nagging problem for us and GitHub wasn't very keen on lifting the [diff limits](https://help.github.com/en/articles/limits-for-viewing-content-and-diffs-in-a-repository#diff-limits).
